---
title: "ATAC-seq Samtools stats alignment quality analysis - split by chromosome"
author: "CIG @ MGI"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
# Set seed
set.seed(0)
rm(list=ls())

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/juanmacias/Library/CloudStorage/Box-Box/LawsonLab/JuanMacias/Postdoc/HPRC/ATACseq/AutomaticTesting/Latest/AlignmentQuality_Analysis/")


# Load packages
library(ggplot2)
require(cowplot)
library(viridis)
library(tidyr)
library(dplyr)
library(ggpubr)
library(pillar)

options(pillar.sigfig = 6)
theme_set(theme_cowplot())

# Set theme object
CS.THEME<-theme(
  panel.background = element_rect(fill="white"),
  panel.grid.major=element_line(color="gray90", linetype="solid"), #adds major grid lines
  panel.grid.minor=element_line(color="gray95", linetype="solid", size=0.65), #adds minor grid lines
  panel.border=element_rect(fill=NA, color="black", size=1.5, linetype="solid"), #draws a black border around plot
  axis.line=element_blank(),
  text=element_text(colour = "black", family= "Arial")
)
```

```{r define input function, include=FALSE}
read_stats <- function(file_path, sample=NA, replicate=NA, assembly=NA, annotations=NA, chromosome=NA) {
  # Read the file
  lines <- readLines(file_path)
  
  # Remove comments from each line
  lines <- sapply(lines, function(line) {
    # Remove everything after and including the '#'
    line <- strsplit(line, "#")[[1]][1]
    # Trim any extra whitespace
    trimws(line)
  })
  
  # Filter out any empty lines
  lines <- lines[lines != ""]
  
  # Split the lines into key-value pairs
  stats <- sapply(lines, function(line) {
    parts <- strsplit(line, ":")[[1]]
    key <- trimws(parts[1])
    value <- trimws(parts[2])
    return(c(key, value))
  })
  
  # Convert to a data frame
  stats_df <- as.data.frame(t(stats), stringsAsFactors = FALSE)
  colnames(stats_df) <- c("Metric", "Value")
  
  # Convert numeric values to numeric type
  stats_df$Value <- as.numeric(stats_df$Value)
  rownames(stats_df)<-NULL
  
  # Add sample, replicate, assembly, and annotations columns if provided
  if (!is.na(sample)) {
    stats_df$Sample <- sample
  }
  if (!is.na(replicate)) {
    stats_df$Replicate <- replicate
  }
  if (!is.na(assembly)) {
    stats_df$Assembly <- assembly
  }
  if (!is.na(annotations)) {
    stats_df$Annotations <- annotations
  }
  if (!is.na(chromosome)) {
    stats_df$Chromosome <- chromosome
  }
  
  return(stats_df)
}
```

```{r read in linear genome runs and format, include=TRUE}
# Read in list of all files in the directory with full path
LinearGenome_file_list <- dir("/Users/juanmacias/Library/CloudStorage/Box-Box/LawsonLab/JuanMacias/Postdoc/HPRC/ATACseq/AutomaticTesting/Latest/AlignmentQuality_Analysis/LinearGenomesAlignment_SamtoolsStats/AlignmentQualityMetricsSplit", full.names = TRUE)

# start to extract run information from file name
LinearGenome_Sample_Information<-unlist(lapply(LinearGenome_file_list, function(X) strsplit(X,"/")[[1]][17]))

# remove "step2.1_trimed_", ".alignmentQualityMetrics", ".LOCAL", and ".wMito" from LinearGenome_Sample_Information
LinearGenome_Sample_Information <- gsub("step2.1_trimed_", "", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub(".alignmentQualityMetrics", "", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub(".LOCAL", "", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub(".wMito", "", LinearGenome_Sample_Information)

# replace "NoBlackList" with "nbl"
LinearGenome_Sample_Information <- gsub(".NoBlackList", " nbl", LinearGenome_Sample_Information)

# replaced any Assembly values containing .maternal with the name Maternal.
LinearGenome_Sample_Information <- gsub("HG00621.maternal", "Maternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG00741.maternal", "Maternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG01952.maternal", "Maternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG01978.maternal", "Maternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG03516.maternal", "Maternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG00621.paternal", "Paternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG00741.paternal", "Paternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG01952.paternal", "Paternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG01978.paternal", "Paternal", LinearGenome_Sample_Information)
LinearGenome_Sample_Information <- gsub("HG03516.paternal", "Paternal", LinearGenome_Sample_Information)

# replace "." with "_"
LinearGenome_Sample_Information <- gsub("\\.", "_", LinearGenome_Sample_Information)

# split the string by "_" to make columns: Sample, Replicate, Assembly and turn into a data frame
LinearGenome_Sample_Information <- as.data.frame(do.call(rbind, strsplit(LinearGenome_Sample_Information, "_")))
colnames(LinearGenome_Sample_Information) <- c("Sample", "Replicate", "Assembly", "Chromosome")

# Check to name sure number of rows in LinearGenome_Sample_Information is the same as the number of files in LinearGenome_file_list
if (nrow(LinearGenome_Sample_Information) != length(LinearGenome_file_list)) {
  stop("Number of files does not match number of rows in LinearGenome_Sample_Information")
} else {
  print("Number of files matches number of rows in LinearGenome_Sample_Information")
}

# Import data and build data frame
lapply(
  1:length(LinearGenome_file_list),
  function(X) read_stats(LinearGenome_file_list[X], sample=LinearGenome_Sample_Information[X,]$Sample, replicate=LinearGenome_Sample_Information[X,]$Replicate, assembly=LinearGenome_Sample_Information[X,]$Assembly, chromosome=LinearGenome_Sample_Information[X,]$Chromosome)
  ) %>% bind_rows() -> LinearGenome_Stats

# Replace " " with "_" in the Metric column
LinearGenome_Stats$Metric <- gsub(" ", "_", LinearGenome_Stats$Metric)

# Spread the data frame to make it wide Metrics as columns Values as fill
spread(LinearGenome_Stats, key=Metric, value=Value) -> LinearGenome_Stats_wide
```

```{r read in pangenome runs and format, include=TRUE}
# Read in list of all files in the directory with full path
Pangenome_file_list <- dir("/Users/juanmacias/Library/CloudStorage/Box-Box/LawsonLab/JuanMacias/Postdoc/HPRC/ATACseq/AutomaticTesting/Latest/AlignmentQuality_Analysis/PangenomeAlignment_SamtoolsStats/AlignmentQualityMetricsSplit", full.names = TRUE)

# start to extract run information from file name
Pangenome_Sample_Information<-unlist(lapply(Pangenome_file_list, function(X) strsplit(X,"/")[[1]][17]))

# remove "_NMadded.alignmentQualityMetrics" from Pangenome_Sample_Information
Pangenome_Sample_Information <- gsub(".alignmentQualityMetrics", "", Pangenome_Sample_Information)

# replace "pangenome.hg38" with "pangenome"
Pangenome_Sample_Information <- gsub("pangenome.hg38", "pangenome", Pangenome_Sample_Information)

# remove "GRCh38#0#" from Pangenome_Sample_Information
Pangenome_Sample_Information <- gsub("GRCh38#0#", "", Pangenome_Sample_Information)

# Fix chromosome names
Pangenome_Sample_Information <- paste(unlist(lapply(Pangenome_Sample_Information, function(X) strsplit(X, "\\.")[[1]][1])),gsub("_",".",unlist(lapply(Pangenome_Sample_Information, function(X) strsplit(X, "\\.")[[1]][2]))),sep = "_")

# split the string by "_" to make columns: Sample, Replicate, Assembly, Chromosome and turn into a data frame
Pangenome_Sample_Information <- as.data.frame(do.call(rbind, strsplit(Pangenome_Sample_Information, "_")))
colnames(Pangenome_Sample_Information) <- c("Sample", "Replicate", "Assembly", "Chromosome")

# Check to name sure number of rows in Pangenome_Sample_Information is the same as the number of files in Pangenome_file_list
if (nrow(Pangenome_Sample_Information) != length(Pangenome_file_list)) {
  stop("Number of files does not match number of rows in Pangenome_Sample_Information")
} else {
  print("Number of files matches number of rows in Pangenome_Sample_Information")
}

# Import data and build data frame
lapply(
  1:length(Pangenome_file_list),
  function(X) read_stats(Pangenome_file_list[X], sample=Pangenome_Sample_Information[X,]$Sample, replicate=Pangenome_Sample_Information[X,]$Replicate, assembly=Pangenome_Sample_Information[X,]$Assembly, chromosome=Pangenome_Sample_Information[X,]$Chromosome)
  ) %>% bind_rows() -> Pangenome_Stats

# Replace " " with "_" in the Metric column
Pangenome_Stats$Metric <- gsub(" ", "_", Pangenome_Stats$Metric)

# Simplify the chromosome names, such that all non primary chromosomes are grouped together
#Pangenome_Stats$Chromosome[!(Pangenome_Stats$Chromosome %in% c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY", "chrM"))]<-"Other"

# Filter out non primary chromosomes
Pangenome_Stats<-subset(Pangenome_Stats, Chromosome %in% c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY", "chrM"))


# Spread the data frame to make it wide Metrics as columns Values as fill
spread(Pangenome_Stats, key=Metric, value=Value, fill = NA) -> Pangenome_Stats_wide
```

```{r combine all data, include=FALSE}
# Combine all data
all_data.long <- rbind(LinearGenome_Stats, Pangenome_Stats)
all_data.wide <- rbind(LinearGenome_Stats_wide, Pangenome_Stats_wide)
```

```{r perform ANOVA for each metric, include=TRUE}
for (METRIC in unique(all_data.long$Metric)){
  print(METRIC)
  # add empty line
  print("")
  TEST<-subset(all_data.long, Metric==METRIC)
  try(
    print(summary(aov(Value~Assembly*Sample, TEST)))
  )
}
```


```{r plot Metrics, include=FALSE}
# Metrics to plot:
# mismatches
# error_rate
# outward_oriented_pairs
# pairs_with_other_orientation
# pairs_on_different_chromosomes
# percentage_of_properly_paired_reads_(%)
# insert_size_average
# insert_size_standard_deviation

# Plot the data - mismatches
Plot_mismatches<-ggplot(all_data.wide, aes(x=Assembly, y=mismatches/1000000))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Mismatches", x="Sample", y="mismatches (M)\n")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "bottom")+
  facet_wrap(~Chromosome, scales="free")

# Extract legend
legendplot<-as_ggplot(get_legend(Plot_mismatches))

# Remove legend from plot
Plot_mismatches<-Plot_mismatches+theme(legend.position="none")

# Plot the data - error_rate
Plot_error_rate<-ggplot(all_data.wide, aes(x=Assembly, y=100*error_rate))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Error rate", x="Sample", y="% error rate\n")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - outward_oriented_pairs
Plot_outward_oriented_pairs<-ggplot(all_data.wide, aes(x=Assembly, y=outward_oriented_pairs/1000000))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Outward oriented pairs", x="Sample", y="outward oriented\npairs (M)")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - pairs_with_other_orientation
Plot_pairs_with_other_orientation<-ggplot(all_data.wide, aes(x=Assembly, y=pairs_with_other_orientation/1000))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Pairs with other orientation", x="Sample", y="pairs with\nother orientation (K)")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - pairs_on_different_chromosomes
Plot_pairs_on_different_chromosomes<-ggplot(all_data.wide, aes(x=Assembly, y=pairs_on_different_chromosomes/1000))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Pairs on different chromosomes", x="Sample", y="pairs on\ndifferent chromosomes (K)")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - percentage_of_properly_paired_reads_(%)
Plot_percentage_of_properly_paired_reads<-ggplot(all_data.wide, aes(x=Assembly, y=`percentage_of_properly_paired_reads_(%)`))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Percentage of properly paired reads (%)", x="Sample", y="%properly paired\nreads")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - insert_size_average
Plot_insert_size_average<-ggplot(all_data.wide, aes(x=Assembly, y=insert_size_average))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Insert size average", x="Sample", y="insert size\naverage")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Plot the data - insert_size_standard_deviation
Plot_insert_size_standard_deviation<-ggplot(all_data.wide, aes(x=Assembly, y=insert_size_standard_deviation))+
  geom_boxplot(outliers = FALSE, color="darkgrey")+
  geom_jitter(width=0.2, aes(color=Sample, shape=Sample))+
  scale_color_viridis(discrete=TRUE, begin=0.1 ,end = 0.9, option = "plasma")+
  labs(title="Insert size standard deviation", x="Sample", y="insert size\nstandard deviation")+
  CS.THEME+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),legend.position = "none")+
  facet_wrap(~Chromosome, scales="free")

# Combine all plots
CombinedPlot<-plot_grid(Plot_mismatches, Plot_error_rate, Plot_outward_oriented_pairs, Plot_pairs_with_other_orientation, Plot_pairs_on_different_chromosomes, Plot_percentage_of_properly_paired_reads, Plot_insert_size_average, Plot_insert_size_standard_deviation, ncol=2, labels = "AUTO")

CombinedPlot<-plot_grid(CombinedPlot, legendplot, ncol=1, rel_heights = c(1, 0.05))
```

```{r plot combined plot, include=TRUE, fig.width=55, fig.height=55}
ggsave2("AlignmentQualityMetrics_SplitByChromosome.png", CombinedPlot, width=55, height=55, dpi=300, limitsize=FALSE)
CombinedPlot
```

```{r calculate summary statistics, include=TRUE}
# I will calculate per group summary statistics for each metric
# Metrics to calulate summary metrics for:
# mismatches
# error_rate
# outward_oriented_pairs
# pairs_with_other_orientation
# pairs_on_different_chromosomes
# percentage_of_properly_paired_reads_(%)
# insert_size_average
# insert_size_standard_deviation

# Calculate summary statistics per sample by chromosome
all_data.wide %>%
  group_by(Sample, Chromosome) %>%
  summarise(
    mismatches_mean=mean(mismatches/1000000),
    mismatches_sd=sd(mismatches/1000000),
    error_rate_mean=mean(100*error_rate),
    error_rate_sd=sd(100*error_rate),
    outward_oriented_pairs_mean=mean(outward_oriented_pairs/1000000),
    outward_oriented_pairs_sd=sd(outward_oriented_pairs/1000000),
    pairs_with_other_orientation_mean=mean(pairs_with_other_orientation/1000),
    pairs_with_other_orientation_sd=sd(pairs_with_other_orientation/1000),
    pairs_on_different_chromosomes_mean=mean(pairs_on_different_chromosomes/1000),
    pairs_on_different_chromosomes_sd=sd(pairs_on_different_chromosomes/1000),
    percentage_of_properly_paired_reads_mean=mean(`percentage_of_properly_paired_reads_(%)`),
    percentage_of_properly_paired_reads_sd=sd(`percentage_of_properly_paired_reads_(%)`),
    insert_size_average_mean=mean(insert_size_average),
    insert_size_average_sd=sd(insert_size_average),
    insert_size_standard_deviation_mean=mean(insert_size_standard_deviation),
    insert_size_standard_deviation_sd=sd(insert_size_standard_deviation)
  ) -> summary_stats

# Write summary statistics to file
write.table(summary_stats, "AlignmentQualityMetrics_SummaryStatistics_SplitByChromosome.csv", row.names=FALSE, quote = FALSE, sep = "\t")

# Calculate summary statistics per assembly by chromosome
all_data.wide %>%
  group_by(Assembly, Chromosome) %>%
  summarise(
    mismatches_mean=mean(mismatches/1000000),
    mismatches_sd=sd(mismatches/1000000),
    error_rate_mean=mean(100*error_rate),
    error_rate_sd=sd(100*error_rate),
    outward_oriented_pairs_mean=mean(outward_oriented_pairs/1000000),
    outward_oriented_pairs_sd=sd(outward_oriented_pairs/1000000),
    pairs_with_other_orientation_mean=mean(pairs_with_other_orientation/1000),
    pairs_with_other_orientation_sd=sd(pairs_with_other_orientation/1000),
    pairs_on_different_chromosomes_mean=mean(pairs_on_different_chromosomes/1000),
    pairs_on_different_chromosomes_sd=sd(pairs_on_different_chromosomes/1000),
    percentage_of_properly_paired_reads_mean=mean(`percentage_of_properly_paired_reads_(%)`),
    percentage_of_properly_paired_reads_sd=sd(`percentage_of_properly_paired_reads_(%)`),
    insert_size_average_mean=mean(insert_size_average),
    insert_size_average_sd=sd(insert_size_average),
    insert_size_standard_deviation_mean=mean(insert_size_standard_deviation),
    insert_size_standard_deviation_sd=sd(insert_size_standard_deviation)
  ) -> summary_stats_assembly

# Write summary statistics to file
write.table(summary_stats_assembly, "AlignmentQualityMetrics_SummaryStatistics_SplitByChromosome_Assembly.csv", row.names=FALSE, quote = FALSE, sep = "\t")

```

```{r print summary statistics by assembly, include=TRUE}
# print summary statistics by assembly for each metric one by one
print("Summary statistics by assembly")

# mismatches
summary_stats_assembly[,c("Assembly", "mismatches_mean")]
summary_stats_assembly[,c("Assembly", "mismatches_sd")]

# error_rate
summary_stats_assembly[,c("Assembly", "error_rate_mean")]
summary_stats_assembly[,c("Assembly", "error_rate_sd")]

# outward_oriented_pairs
summary_stats_assembly[,c("Assembly", "outward_oriented_pairs_mean")]
summary_stats_assembly[,c("Assembly", "outward_oriented_pairs_sd")]

# pairs_with_other_orientation
summary_stats_assembly[,c("Assembly", "pairs_with_other_orientation_mean")]
summary_stats_assembly[,c("Assembly", "pairs_with_other_orientation_sd")]

# pairs_on_different_chromosomes
summary_stats_assembly[,c("Assembly", "pairs_on_different_chromosomes_mean")]
summary_stats_assembly[,c("Assembly", "pairs_on_different_chromosomes_sd")]

# percentage_of_properly_paired_reads_(%)
summary_stats_assembly[,c("Assembly", "percentage_of_properly_paired_reads_mean")]
summary_stats_assembly[,c("Assembly", "percentage_of_properly_paired_reads_sd")]

# insert_size_average
summary_stats_assembly[,c("Assembly", "insert_size_average_mean")]
summary_stats_assembly[,c("Assembly", "insert_size_average_sd")]

# insert_size_standard_deviation
summary_stats_assembly[,c("Assembly", "insert_size_standard_deviation_mean")]
summary_stats_assembly[,c("Assembly", "insert_size_standard_deviation_sd")]
```



